adoc/extensions/sycl_khr_group_interface.adoc

```diff
@@ -0,0 +1,849 @@
+[[sec:khr-group-interface]]
+= sycl_khr_group_interface
+
+This extension provides an alternative interface for groups of work-items
+(including work-groups, sub-groups, and individual work-items) that is simpler
+and less verbose than the interface provided by `sycl::group` and
+`sycl::sub_group` in SYCL 2020.
+
+[[sec:khr-group-interface-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+Some features of this extension are only available when a SYCL implementation
+conforms to {cpp23} or later.
+
+[[sec:khr-group-interface-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_GROUP_INTERFACE# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+[[sec:khr-group-interface-common]]
+== Common group interface
+
+The `khr::work_group`, `khr::sub_group` and `khr::member_item` objects defined
+by this extension all implement a common set of operations, which are shown in
+the synopsis below.
+The name [code]#+__group__+# in that synopsis is a placeholder for each of these
+three types.
+When the synopsis shows an ellipsis ([code]#+/*...*/+#), the subsequent sections
+clarify the definition for each type.
+
+Whether it is possible to query certain properties of a group at compile-time
+depends on the group's type.
+A member function that is not declared [code]#constexpr# in the synopsis below
+may still be declared [code]#constexpr# for a specific group type defined in a
+subsequent section.
+
+[source,role=synopsis]
+----
+
+namespace sycl::khr {
+
+class __group__ {
+
+ public:
+  using id_type = /* ... */;
+  using linear_id_type = /* ... */;
+  using range_type = /* ... */;
+  using extents_type = /* ... */; // C++23
+  using size_type = /* ... */;
+  static constexpr int dimensions = /* ... */;
```

Comment:
If we're looking to align with ISO C++ should we consider calling this `rank` instead?

Reply:
There is a `static constexpr rank()` function that's available starting with C++23.

`dimensions` is already part of the SYCL 2020 group interface (in the sense that it's defined by `group` and `sub_group`). 
 I assumed we wouldn't want to remove that any time soon, so made it part of this new interface as well.

Reply:
Yeah that's a good point, we don't want to break with the existing interface, though perhaps we could have an additional value rank which is equal to `dimensions`, and then we could look to deprecate `dimensions` in the future.

Though saying that, this is not really an issue, just that it would be nice to align with ISO C++ naming, so it may not be worth it.